In [7]:

import pandas as pd
import numpy as np 
import folium
from folium.plugins import MarkerCluster

df = pd.read_csv('AB_NYC_2019.csv')

def generateBaseMap(default_location=[40.693943, -73.985880]):
    base_map = folium.Map(location=default_location)
    return base_map

base_map = generateBaseMap()
folium.plugins.HeatMap(data=df[['latitude', 'longitude', 'price']].
        groupby(['latitude', 'longitude']).mean().
        reset_index().values.tolist(), radius=8, max_zoom=13).add_to(base_map)
base_map.save('index.html')



In [33]:
#For data manipulation
import pandas as pd
import numpy as np 


from sklearn.cluster import KMeans
from sklearn import metrics 
#For scaling certain parameters
from sklearn.preprocessing import MinMaxScaler
from sklearn.preprocessing import StandardScaler
import matplotlib.pyplot as graph



#read the dataset
df = pd.read_csv('AB_NYC_2019.csv')

#remove all the row with the missing data values
clean_df = df.dropna()

#print(clean_df.columns)
#print(clean_df.dtypes['price'])

#Converting the price column from int to float
clean_df['price'] = clean_df['price'].astype(float)

#scaling the input parameters of the model
#scale = MinMaxScaler()
#clean_df[['latitude','longitude','price']] = scale.fit_transform[clean_df['latitude','longitude','price']]
scale_columns = ['latitude' , 'longitude','price']
features = clean_df[scale_columns]
scaler = StandardScaler().fit(features.values)
features = scaler.transform(features.values)
clean_df[scale_columns] = features





#Set the parameters for Kmeans Clustering
input_parameters = clean_df[['latitude','longitude','price']]
kmeans = KMeans(n_clusters=5, init='k-means++',n_init=10).fit(input_parameters)
cluster_labels = kmeans.labels_
#get the centres back from k_means
centers = kmeans.cluster_centers_
#Get the raw or unscled data back from the Scaled data
centers = scaler.inverse_transform(centers)



print("********************************")
print("QUESTION 3C")
print("REPORTING THE AVERAGE PRICE OF NEW YORK STAYS")
print("********************************")
print("CLUSTER   AVERAGE PRICE PER NIGHT")

#Print the average prices found for differnet clusters
for i in range(5):
    print(i ,end = "               " )
    print("$" , int(centers[i][2]))
    





<ipython-input-33-47a3ec3cb5dc>:25: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  clean_df['price'] = clean_df['price'].astype(float)
<ipython-input-33-47a3ec3cb5dc>:34: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  clean_df[scale_columns] = features
/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexing.py:1738: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: h

********************************
QUESTION 3C
REPORTING THE AVERAGE PRICE OF NEW YORK STAYS
********************************
CLUSTER   AVERAGE PRICE PER NIGHT
0               $ 90
1               $ 108
2               $ 223
3               $ 99
4               $ 6439


The average Price range of staying in New York ranges from $99-$6439 per night. The average price range of all the clusters is in the expected range except for thr 4the Cluster which is turns out to be on the upper side on New York. This can be confirmed by looking at the geographic location of this cluster on the map